pyV2DL3/eventdisplay/fillRESPONSE.py

```diff
@@ -37,53 +37,77 @@ def print_logging_info(irf_to_store, camera_offsets, pedvar, zenith):
-    fuzzy_boundary = clk.params["fuzzy_boundary"]
-    if np.all(par_irf < par) or np.all(par_irf > par):
-        if not force_extrapolation:
-            if fuzzy_boundary > 0.0:
-                fuzzy_diff_u = 1 - np.max(par_irf) / par
-                if fuzzy_diff_u < fuzzy_boundary:
-                    logging.warning(
-                        "Coordinate difference is {0:0.3f} of upper boundary value and is within {1:0.3f}".format(
-                            fuzzy_diff_u, fuzzy_boundary
-                        )
-                    )
-                    par = np.max(par_irf)
-                elif np.min(par_irf) != 0:
-                    fuzzy_diff_l = 1 - par / np.min(par_irf)
-                    if fuzzy_diff_l <= fuzzy_boundary:
-                        logging.warning(
-                            "Coordinate difference is {0:0.3f} of lower boundary value and is within {1:0.3f}".format(
-                                fuzzy_diff_l, fuzzy_boundary
-                            )
-                        )
-                        par = np.min(par_irf)
+    tolerance = clk.params["fuzzy_boundary"]
+    if np.all(irf_stored_par < par) or np.all(irf_stored_par > par):
+        if clk.params["force_extrapolation"]:
+            logging.warning("IRF extrapolation allowed for coordinate not inside IRF {0} range".format(par_name))
+        else:
+            if tolerance > 0.0:
+                boundary = np.max(irf_stored_par)
+                if check_fuzzy_boundary(par, boundary, tolerance):
+                    par = np.max(irf_stored_par)
                 else:
-                    raise ValueError("Coordinate not inside IRF {0} range".format(par_name))
+                    if np.min(irf_stored_par) != 0:
+                        boundary = np.min(irf_stored_par)
+                        if check_fuzzy_boundary(par, boundary, tolerance):
+                            par = np.min(irf_stored_par)
             else:
-                raise ValueError("Coordinate not inside IRF {0} range".format(par_name))
-        else:
-            logging.warning("IRF extrapolation allowed for coordinate not inside IRF {0} range".format(par_name))
+                raise ValueError(
+                    "Coordinate not inside IRF {0} range! Try using --fuzzy_boundary".format(par_name)
+                )
     return par
 
 
+def check_fuzzy_boundary(par, boundary, tolerance):
+    """" Checks if the paramters values is within the given tolerance.
+    tolerance parameter is defined as ratio of absolute difference
+    between boundary and par to the boundary.
+
+    Parameters
+    ----------
+    par: parameter of given run, it can be pedvar, zenith or camera offset
+    boundary: lower or upper boundary value of stored IRF
+    tolerance: allowed value of --fuzzy_boundary command line argument
+
+    Returns
+    -------
+    Boolean value if tolerance is within given allowed value
+
+    """
+
+    fuzzy_diff = np.abs(boundary - par) / boundary
```

Comment:
is it possible that boundary is zero? In this case you have to check to prevent division by zero.

Reply:
boundary == 0, happens only for lower boundary, and that is taken care before calling this function for lower boundary value.

Reply:
I would do that in check_fuzzy_boundary(), is this is actually the place where it matters. When reading the code, it is otherwise not clear why you check for `boundary != 0` (and you check for ` np.min(irf_stored_par) != 0`, which is only later assigned to boundary.